### PR TITLE
Implement our own BlockIterator based on bukkit's

### PR DESCRIPTION
**Pull request by KevinDaGame**
## What type of PR is this? (check all applicable)

- [x] Feature
- [ ] Refactor
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Implement BlockIterator

## QA Instructions, Screenshots, Recordings

Tested with linebrush

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
  have not been included_
- [ ] I need help with writing tests
- [x] Not needed

### CODE REVIEW

File: VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/block/BlockFace.java [hunk 1]
```diff
@@ -48,4 +48,28 @@ public int getModZ() {
...
+            case SOUTH_WEST -> NORTH_EAST;
+            case WEST_NORTH_WEST -> EAST_SOUTH_EAST;
+            case NORTH_NORTH_WEST -> SOUTH_SOUTH_EAST;
+            case NORTH_NORTH_EAST -> SOUTH_SOUTH_WEST;
+            case EAST_NORTH_EAST -> WEST_SOUTH_WEST;
+            case EAST_SOUTH_EAST -> WEST_NORTH_WEST;
+            case SOUTH_SOUTH_EAST -> NORTH_NORTH_WEST;
+            case SOUTH_SOUTH_WEST -> NORTH_NORTH_EAST;
+            case WEST_SOUTH_WEST -> EAST_NORTH_EAST;
+            case SELF -> SELF;
```
Comment (Lennart99) [inline VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/block/BlockFace.java:71]: There isn't any way to do this nicer, is there? maybe based on the mod [xyz]?

**Review by Lennart99 — APPROVED** [review]

File: VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/block/BlockFace.java (hunk 1 above)
Comment (KevinDaGame) [reply VoxelSniperCore/src/main/java/com/github/kevindagame/voxelsniper/block/BlockFace.java:71]: Apart from giving the enum an opposite param, I don't think so

File: VoxelSniperForge/src/main/java/com/github/kevindagame/voxelsniperforge/world/ForgeWorld.java [hunk 2]
```diff
@@ -72,13 +76,16 @@ public IChunk getChunkAtLocation(int x, int z) {
 
     @Override
     public List<IEntity> getNearbyEntities(BaseLocation location, double x, double y, double z) {
-        throw new UnsupportedOperationException("Not implemented yet");
+        var aabb = new AABB(location.getX()-x, location.getY()-y, location.getZ()-z, location.getX()+x, location.getY()+y, location.getZ()+z);
+        List<Entity> notchEntityList = this.level.getEntities(null, aabb);
```
[the remark this answers is not in the document]
Comment (Lennart99) [reply VoxelSniperForge/src/main/java/com/github/kevindagame/voxelsniperforge/world/ForgeWorld.java:80]: probably copied it over, will change it now

[the remark this answers is not in the document]
Comment (Lennart99) [reply VoxelSniperForge/src/main/java/com/github/kevindagame/voxelsniperforge/world/ForgeWorld.java]: https://github.com/KevinDaGame/VoxelSniper-Reimagined/pull/181/commits/8af24a0c4fc7c0921994ea2d045ba4b82b684ab2

**Pull request merged**